File: neoforge/src/main/java/FallingTreesNeoForge.java
/*
 * Copyright (c) 2025. Oliver Froberg
 *
 * This code is licensed under the GNU General Public License v3.0
 * See: https://www.gnu.org/licenses/gpl-3.0-standalone.html
 */

package dev.pandasystems.fallingtrees.neoforge;

import dev.pandasystems.fallingtrees.FallingTrees;
import dev.pandasystems.fallingtrees.utils.BlockMapEntityData;
import dev.pandasystems.fallingtrees.utils.ItemListEntityData;
import dev.pandasystems.pandalib.api.registry.deferred.DeferredRegister;
import net.minecraft.network.syncher.EntityDataSerializer;
import net.neoforged.bus.api.IEventBus;
import net.neoforged.fml.common.Mod;
import net.neoforged.neoforge.registries.NeoForgeRegistries;

@Mod(FallingTrees.MOD_ID)
public class FallingTreesNeoForge {
	public final static DeferredRegister<EntityDataSerializer<?>> ENTITY_DATA =
			DeferredRegister.create(FallingTrees.MOD_ID, NeoForgeRegistries.ENTITY_DATA_SERIALIZERS);

    public FallingTreesNeoForge(IEventBus eventBus) {
		FallingTrees instance = FallingTrees.INSTANCE;

		ENTITY_DATA.register("block_map", key -> BlockMapEntityData.BLOCK_MAP);
		ENTITY_DATA.register("item_list", key -> ItemListEntityData.ITEM_LIST);
		ENTITY_DATA.register();
	}
}

File: neoforge/src/main/kotlin/FallingTreesNeoForge.kt
/*
 * Copyright (C) 2025-2025 Oliver Froberg (The Panda Oliver)
 *
 * This program is free software: you can redistribute it and/or modify
 *  it under the terms of the GNU General Public License as published by
 *  the Free Software Foundation, either version 3 of the License, or
 *  any later version.
 *
 * You should have received a copy of the GNU General Public License
 *  along with this program. If not, see <https://www.gnu.org/licenses/>.
 */
package dev.pandasystems.fallingtrees.neoforge

import dev.pandasystems.fallingtrees.FallingTrees
import net.neoforged.bus.api.IEventBus
import net.neoforged.fml.common.Mod

@Mod(FallingTrees.modid)
class FallingTreesNeoForge(eventBus: IEventBus) {
	init {
		FallingTrees
	}
}
